source/event_stream.c

```diff
@@ -288,6 +299,8 @@ int aws_event_stream_message_init(
     struct aws_allocator *alloc,
     struct aws_array_list *headers,
     struct aws_byte_buf *payload) {
+    AWS_FATAL_PRECONDITION(message);
+    AWS_FATAL_PRECONDITION(alloc);
 
     size_t payload_len = payload ? payload->len : 0;
```

Comment:
can't comment on it from this PR, but below this line we call a function to calculate `headers_length` and that function can have integer overflow

Reply:
cool. will track down.